day-2

maximASH21 opened the pull request:
day2
![image](https://user-images.githubusercontent.com/86965986/125251390-e9ae8a80-e2ff-11eb-842c-f4a84d83d767.png)

maximASH21 committed:
upd app.js ,game.html

alinaaZ committed:
! Info about branches

Добавили файл с навигацией по существующим веткам

maximASH21 committed:
upd

upd

alinaaZ committed:
Upd info about branches

alinaaZ commented:
Так как у нас были проблемы с ветками, чтобы никто не запутался, мы создали файл с навигацией по веткам - branches info.txt. 

![image](https://user-images.githubusercontent.com/87071309/125077428-c3070e80-e0c9-11eb-958b-6d6eb76d1657.png)

alinaaZ committed:
Update readme.md

alinaaZ committed:
Upd app.js & game.html (Task 2)

Task 2. Опишите функцию которая задаст размеры игрового поля

alinaaZ committed:
Upd app.js (Task1.2)

Task1.2 Объявление переменных и их связка с DOM (часть)

maximASH21 committed:
task 4

complete task4

maximASH21 committed:
upd app.js

подправил синтаксис

maximASH21 committed:
upd game.html

убрал штуку для теста

maximASH21 committed:
UPD

UPD

VictorKashtanov committed:
Update app.js

task 1.1 дополнение

VictorKashtanov committed:
Update game.html

изменения game.html для task 1.1

alinaaZ committed:
upd app.js & game.html (task 1.2)

Добавлены недостающие кнопки + Завершенное задание Task1.2 Объявление переменных и их связка с DOM

VictorKashtanov committed:
Update game.html

task 1.1 fix

alinaaZ committed:
Upd app.js & game.html

alinaaZ committed:
Upd app.js&game.html&style.css

for task 1.1

alinaaZ committed:
upd app.js&game.html (task 1.1)

for task 1.1. again

VictorKashtanov committed:
app.js and game.html

1.1 update

VictorKashtanov committed:
Merge branch 'day-2' of https://github.com/praktika-organiz-cat/rd-summer-practice-2021-students into day-2

alinaaZ committed:
upd app.js&game.html (task 1.1)

for task 1.1

lyolka92 commented on src/app.js:
```diff
@@ -758,56 +758,75 @@
-                    // $mapCanvas: ,
-                    // mapCellSize: 25
+                    
+                   $gameCaption: $('gameCaption'),
```
Когда вы пишите селектор, нужно чётко понимать по какому признаку вы ищете элемент:
```java-script
$('div') // вернёт элемент с тегом div
$('.div') // вернёт элемент с классом div
$('#div') // вернёт элемент с id 'div'
```

alinaaZ replied:
Точно, забыла # добавить. Исправила, спасибо.

![image](https://user-images.githubusercontent.com/87071309/125162384-2e1b1880-e190-11eb-92fc-f565f60d4eb7.png)

lyolka92 commented on src/app.js:
```diff
@@ -758,56 +758,75 @@
+                        },
+                       
+                     mapBuffer: null,
+                     $mapCanvas ,
```
Этот элемент тоже нужно взять из html - это тот самый canvas, где будет происходить сама игра

alinaaZ replied:
Добавила.

![image](https://user-images.githubusercontent.com/87071309/125162403-43904280-e190-11eb-87db-a13311817b5e.png)

lyolka92 commented on src/app.js:
```diff
@@ -758,56 +758,75 @@
+                <button id="btnReconnect" class="btn-dialog"onclick="">Перезапуск</button> -->
+                <button id="btnCancel" class="btn-warn" onclick="">Отменить игру</button>
+                <button id="btnExit" class="btn-warn" onclick="">Выход</button> */
+
```
Тут и далее - комментарии, которые не являются подсказкой к логике работы кода, мы обычно не оставляем в коде.

alinaaZ replied:
Ок, убрала

lyolka92 commented on src/game.html:
```diff
@@ -12,63 +12,152 @@
             </div>
             <div class="center">
-                <canvas id="map" class="map"></canvas>
+                <canvas id="map" class="map" width=​"300" height="300"></canvas>
```
Тут ширину и высоту можно убрать

alinaaZ replied:
Хорошо, спасибо. Убрала

alinaaZ committed:
upd app.js & game.html

Исправление ошибок (task 1.1, 1.2, 2)

alinaaZ committed:
upd app.js (task 3.1)

for Task 3.1 обработчики событий

alinaaZ committed:
Update game.html

alinaaZ committed:
upd app.js (task 3.1)

alinaaZ committed:
Update app.js

tasks 9, 10, 11

alinaaZ committed:
Update style.css

подсветка

alinaaZ committed:
upd game.html&style.css

maximASH21 committed:
task 3 comlete

update

VictorKashtanov committed:
task 5

task 5 and html upd

maximASH21 committed:
upd todo4

upd

maximASH21 committed:
task 8 upd

task 8 upd

maximASH21 committed:
upd task 6,7

update task 6.7

alinaaZ committed:
updates

alinaaZ committed:
upd

alinaaZ committed:
upd task3

alinaaZ committed:
updates

попытки привести все в нормальный вид

alinaaZ committed:
upd style.css

карта по центру

alinaaZ committed:
upd style.css&game.html

alinaaZ committed:
upd game-timer

lyolka92 commented on src/app.js:
```diff
@@ -880,41 +874,84 @@
                 // var c = this.state.callbacks;
+                var c = this.state.callbacks;
                 // c.captionChanged
+                c.captionChanged.add(function (name, status){this.setGameCaption(name, status);}.bind(this))
```
Тут и далее - можно проще:
```javascript
c.captionChanged.add(this.setGameCaption.bind(this));
```
Нужные параметры в функцию передадутся.

lyolka92 commented on src/app.js:
```diff
@@ -880,41 +874,84 @@
+
+                btns.$btnGameList.click(function () {
+                    window.location.href = 'index.html';
+                }.bind(this));
```
Тут bind не нужен - у вас в функции же не используется this

lyolka92 commented on src/app.js:
```diff
@@ -880,41 +874,84 @@
+                }.bind(this));
+                btns.$btnStart.click(function () {
+                    this.state.game.start();
+                }.bind(this));
```
Тут и далее - можно в function не оборачивать
```javascript
btns.$btnStart.click(this.state.game.start().bind(this));
```

lyolka92 commented on src/app.js:
```diff
@@ -1148,16 +1186,28 @@
                 /**
                  * TODO: Task 9. Опишите доступность элементов при загрузке игры $container $error $loading
                  */
+               /* utils.removeClasses($loading, 'hidden') //тк ShowLoading
```
Это и следующие два задания правильно выполнены. Раскомментируйте их только)

lyolka92 reviewed (commented):
Общее замечание - уберите из кода лишние комментарии + доделать невыполненные задания

alinaaZ committed:
mini updates

alinaaZ committed:
upd

alinaaZ committed:
final updates tasks 9-11

maximASH21 committed:
update task 3 .

добавлено состояние для отменённой игры

maximASH21 committed:
complete task 8

complete task 8

alinaaZ commented:
Состояние с выполненными заданиями 1-11. Игра работает

![image](https://user-images.githubusercontent.com/87071309/125235576-59664a80-e2eb-11eb-97bf-4139515ac5d7.png)

alinaaZ committed:
final tasks 1-11

Убрала лишние комментарии

lyolka92 commented on src/app.js:
```diff
@@ -1020,7 +1007,14 @@
+                if (!img) {
+                    console.log(img, this.imgs, police, self, this.imgs.imgPoliceSelf, this.imgs.imgPolice, 
+                        this.imgs.imgThiefSelf, this.imgs.imgThief);
+                        debugger;
```
Деббагеры и console.log мы не оставляем в продавшей-коде
